llvm/docs/ProgrammersManual.rst

```diff
@@ -2779,17 +2761,9 @@ obtaining it via iteration over some structure:
     if (it != inst->getParent()->end()) errs() << *it << "\n";
   }
 
-Unfortunately, these implicit conversions come at a cost; they prevent these
-iterators from conforming to standard iterator conventions, and thus from being
-usable with standard algorithms and containers.  For example, they prevent the
-following code, where ``B`` is a ``BasicBlock``, from compiling:
-
-.. code-block:: c++
-
-  llvm::SmallVector<llvm::Instruction *, 16>(B->begin(), B->end());
-
-Because of this, these implicit conversions may be removed some day, and
-``operator*`` changed to return a pointer instead of a reference.
+..
+  The part about implicit conversion from an iterator to a pointer is out of
+  date and has been removed.
```

Comment:
I added these comments to let someone blame me in the future.

Reply:
why not just delete it then?

Reply:
> why not just delete it then?

When I went through the commit history to confirm the commit that removed `operator pointer()`, I used GitHub blame to find out who modified the code. I found that commits solely removing content are particularly difficult to locate, as they do not appear in GitHub blame. Therefore, I believe it's best for me to leave a comment here to ensure that someone in the future can blame me in case these revisions cause issues.

Do you think this is a good idea? Let me know your opinion.

Reply:
That's considerate, but I agree with @nickdesaulniers that we don't need to be so cautious. Let's just describe the current state accurately. (If we documented all the things we removed we wouldn't have space for the stuff that's still here!)